File: Reservamos/app/src/main/java/io/github/pabloeferreyra/reservamos/MainActivity.java
package io.github.pabloeferreyra.reservamos;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.MalformedURLException;
import java.net.URL;

import javax.net.ssl.HttpsURLConnection;

public class MainActivity extends AppCompatActivity {

    private static int SPLASH_TIMEOUT = 5000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent homeIntent = new Intent(MainActivity.this, HomeActivity.class);
                startActivity(homeIntent);
                finish();
            }
        }, SPLASH_TIMEOUT);
        AsyncTask.execute(new Runnable() {
            @Override
            public void run() {
                URL githubEndpoint = null;
                HttpsURLConnection myConnection = null;
                try {
                    githubEndpoint = new URL("https://api.github.com/");
                } catch (MalformedURLException e) {
                    e.printStackTrace();
                }
                try {
                    myConnection =
                            (HttpsURLConnection) githubEndpoint.openConnection();
                } catch (IOException e) {
                    e.printStackTrace();
                }

                myConnection.setRequestProperty("User-Agent", "my-rest-app-v0.1");
                try {
                    if (myConnection.getResponseCode() == 200) {
                        InputStream responseBody = myConnection.getInputStream();
                        InputStreamReader responseBodyReader =
                                new InputStreamReader(responseBody, "UTF-8");
                    } else {
                        // Error handling code goes here
                    }
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        });
    }
}

File: Reservamos/app/src/main/java/io/github/pabloeferreyra/reservamos/MainActivity.kt
package io.github.pabloeferreyra.reservamos

import android.app.Activity
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.ListView
import android.widget.Toast
import io.github.pabloeferreyra.reservamos.Adapters.LocalesAdapter
import io.github.pabloeferreyra.reservamos.Entity.Locales

import org.java_websocket.client.WebSocketClient

class MainActivity : AppCompatActivity() {

    private var mWebSocketClient: WebSocketClient? = null
    private var viewParts: Runnable? = null
    private var m_adapter: LocalesAdapter? = null
    private var listaLocales : ArrayList<Locales?> = ArrayList<Locales?>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val lv: ListView = findViewById(R.id.mainListView)
        val ser: Servicio = Servicio()
        listaLocales = ser.getData()
        m_adapter = LocalesAdapter(this, listaLocales)
        lv.adapter = m_adapter
    }



    fun Activity.toast(message: CharSequence, duration: Int = Toast.LENGTH_LONG) {
        Toast.makeText(this, message, duration).show()
    }
}
